core/src/main/java/com/expedia/www/haystack/client/dispatchers/clients/LoggerClient.java

```diff
@@ -0,0 +1,75 @@
+package com.expedia.www.haystack.client.dispatchers.clients;
+
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.expedia.www.haystack.client.Span;
+import com.expedia.www.haystack.client.dispatchers.formats.Format;
+
+public class LoggerClient implements Client {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoggerClient.class);
+
+    private Format<?> format;
+    private Logger logger;
+
+    public LoggerClient(Format<?> format) {
+        this(format, null);
+    }
+
+    public LoggerClient(Format<?> format, Logger logger) {
+        this.format = format;
+        if (logger == null) {
+            this.logger = LoggerFactory.getLogger(this.getClass());
+        } else {
+            this.logger = logger;
+        }
+    }
+
+    @Override
+    public boolean send(Span span) {
+        LOGGER.debug("Received span: {}", span);
+        logger.info("{}", format.format(span));
+        LOGGER.debug("Sent formated span: {}", format.format(span));
+        return true;
+    }
+
+    @Override
+    public void close() throws IOException {
+        // do nothing
+    }
+
+    @Override
+    public void flush() throws IOException {
+        // do nothing
+    }
+
+    public static final class Builder {
+        private Format<?> format;
+        private String loggerName;
+
+        public Builder(Format<?> format) {
```

Comment:
The pattern for Builder that I like the best (I don't remember where I read this, it might have been Effective Java) calls for required arguments to be part of the Builder constructor and for optional arguments to have with...() methods. You've done both here.

Reply:
`Format<?>` is the only required argument; `loggerName` is entirely optional.

Reply:
Then there shouldn't be a withFormat() method and this.format should be final.